pkg/provisioner/ironic/ironic.go

```diff
@@ -1347,6 +1383,10 @@ func (p *ironicProvisioner) Provision(hostConf provisioner.HostConfigData) (resu
 		return transientError(provisioner.NeedsRegistration)
 	}
 
+	if err = p.updateAutomatedClean(ironicNode); err != nil {
```

Comment:
It seems like a mistake to call this in Provision() (or even Deprovision()), since there are numerous opportunities for the node to get cleaned at other times - notably right after registration, but also at any other time when the ironic database is dropped and we don't call Adopt().
I think we should be setting the cleaning mode in ValidateManagementAccess(), which gets called on every reconcile. That way the node is always in a state with the right setting. Perhaps we could consolidate this along with the 3 calls we are already making to nodes.Update() in that method into a single one.

Reply:
I remember a similar [comment](https://github.com/metal3-io/metal3-docs/pull/151#discussion_r527709842) from @dhellmann on the proposal that we could to do it in `ValidateManagementAccess()`, but I did it within `Provision()` and  `Deprovision()` because I thought these are the only methods from after which Ironic node gets cleaned. From Ironic [state machine](https://docs.openstack.org/ironic/pike/_images/states.svg). I see that Ironic cleaning state will come to cleaning state either after managable (BMH Ready state) or active -> deleting (BMH deprovisioning state).
@zaneb Would you mind to share an example use case where Ironic node gets cleaned without us calling `Provision()` or  `Deprovision()` ?

Reply:
You're correct, I misremembered the change in 18465130c8f42bda4aa8191b5093656dcc029d90.
I do plan to change the state machine so that the Preparing state will move the state from Manageable->Available (and thus trigger automatic cleaning) as [proposed on the mailing list](https://groups.google.com/g/metal3-dev/c/N-EpqQn94bw/m/nRkM9bmuAAAJ), but currently it does not do so.

Reply:
I made a similar [comment](https://github.com/metal3-io/baremetal-operator/pull/784#discussion_r594475109) but the current approach works, so perhaps we can address the consolidation of Updates in a follow-up PR?